201-cdn-with-ruleseengine-urlsigning/azuredeploy.json

```diff
@@ -0,0 +1,183 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "profileName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the CDN Profile"
+      }
+    },
+    "endpointName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the CDN Endpoint"
+      }
+    },
+    "originUrl": {
+      "type": "string",
+      "metadata": {
+        "description": "Url of the origin"
+      }
+    },
+    "CDNSku": {
+      "type": "string",
+      "metadata": {
+        "description": "CDN SKU names"
+      },
+      "defaultValue": "Standard_Microsoft",
+      "allowedValues": [
+        "Standard_Akamai",
+        "Standard_Verizon",
+        "Premium_Verizon",
+        "Standard_Microsoft"
+      ]
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    }
+  },
+  "resources": [
+    {
+      "name": "[parameters('profileName')]",
+      "type": "Microsoft.Cdn/profiles",
+      "location": "[parameters('location')]",
+      "apiVersion": "2020-03-31",
+      "sku": {
+        "name": "[parameters('CDNSku')]"
+      },
+      "resources": [
+        {
+          "apiVersion": "2020-03-31",
+          "dependsOn": [
+            "[resourceId('Microsoft.Cdn/profiles', parameters('profileName'))]"
+          ],
+          "location": "[parameters('location')]",
+          "name": "[parameters('endpointName')]",
+          "type": "endpoints",
+          "properties": {
+            "originHostHeader": "[parameters('originUrl')]",
+            "isHttpAllowed": true,
+            "isHttpsAllowed": true,
+            "queryStringCachingBehavior": "IgnoreQueryString",
+            "contentTypesToCompress": [
+              "text/plain",
+              "text/html",
+              "text/css",
+              "application/x-javascript",
+              "text/javascript"
+            ],
+            "isCompressionEnabled": true,
+            "origins": [
+              {
+                "name": "origin1",
+                "properties": {
+                  "hostName": "[parameters('originUrl')]"
+                }
+              }
+            ],
+			"urlSigningKeys": [
+				{
+				  "keyId": "key1",
+				  "keySourceParameters":{
```

Comment:
Seems like these values should be provided by the user?

Reply:
@bmoore-msft , Yes. these should ideally be provided by customer pointing to their keyvault. As the repro try to submit the template to Azure RP, we created a sample key vault and referenced here. Let me know if there are better way to handle this. Users who use the template will not know what they keys are as they don't have permissions. Only Azure CDN APP (Principal) will have access to this key vault.

Reply:
@bmoore-msft , Thanks for the clear ask. I updated the readme file. Please take a look.

Reply:
Hi @bmoore-msft , we are waiting to share this example in our user documentation. Can you please look into this PR?

Reply:
A couple comments on the readme - also, the values below need to be parameters for the template.

Reply:
These still need to be moved to parameters

Reply:
Parametrized the urlsigning keys.